### PR TITLE
client: added "channel" parameter support to InstallSnap and RefreshSnap

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

**Comment by mvo5** [comment]
LGTM

**Comment by mvo5** [comment]
retest this please

File: client/snap_op.go [hunk 1]
```diff
@@ -42,11 +43,12 @@ func (client *Client) RemoveSnap(name string) (string, error) {
 	return client.doAsync("POST", path, nil, body)
 }
 
-// RefreshSnap refreshes the snap with the given name, returning the UUID of the
-// background operation upon success.
-func (client *Client) RefreshSnap(name string) (string, error) {
+// RefreshSnap refreshes the snap with the given name (switching it to track
+// the given channel if given), returning the UUID of the background operation
+// upon success.
```
Comment (niemeyer) [inline client/snap_op.go]: Suggested wording:

```
// RefreshSnap finds out what the most recent version of the named snap is
// according to the provided channel, installs it if not current, and returns an
// identifier to track the operation.
// The channel name is optional and if empty defaults to the last used channel
// for the named snap.
```

File: client/snap_op.go [hunk 2]
```diff
@@ -24,11 +24,12 @@ import (
 	"strings"
 )
 
-// InstallSnap adds the snap with the given name, returning the UUID of the
-// background operation upon success.
-func (client *Client) InstallSnap(name string) (string, error) {
+// InstallSnap adds the snap with the given name from the given channel (or
+// the system default channel if not), returning the UUID of the background
+// operation upon success.
```
Comment (niemeyer) [inline client/snap_op.go]: Suggested wording:
```
// InstallSnap finds out what the most recent version of the named snap is
// according to the provided channel and installs it. The channel name is
// optional and if empty defaults to "stable".
// It returns an identifier to track the background operation.
```

Comment (niemeyer) [inline client/snap_op.go]: Is that actually true?  What are the semantics for missing channels?

**Comment by niemeyer** [comment]
LGTM

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Pull request merged**